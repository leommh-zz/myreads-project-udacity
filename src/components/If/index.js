/**
 * @description - Tests a condition
 * @var {string} test - Parameter to be used in the test
 * @var {component} children - Child component    
 * @returns - If the test is valid it will return the component, else it will return false
 */
const If = ({ test, children }) => (
    test ? children : false
);

export default If;