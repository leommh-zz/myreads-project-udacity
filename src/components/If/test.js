import React from 'react';
import { shallow } from 'enzyme';
import If from './index';

describe('[Component] If', () => {
  
    const children = () => console.log('If Test Children True');

    it('Display If Component test true response', () => {
        const wrapper = shallow(
            <If test={true} children={children} />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('Display If Component test false response', () => {
        const wrapper = shallow(
            <If test={false} children={children} />
        );
        
        expect(wrapper).toMatchSnapshot();
    });
    
});