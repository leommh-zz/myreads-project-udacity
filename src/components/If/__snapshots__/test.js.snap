// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] If Display If Component test false response 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <If
    test={false}
>
    [Function]
</If>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`[Component] If Display If Component test true response 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <If
    test={true}
>
    [Function]
</If>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): [Function],
  Symbol(enzyme.__nodes__): Array [
    [Function],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
